Assets/Scripts/Player/Movement/PlayerMovement.cs

```diff
@@ -1,36 +1,42 @@
 using System;
-using System.Collections;
-using FrameWork.Extensions;
 using FrameWork.GridSystem;
 using UnityEngine;
+using UnityEngine.Events;
 
 namespace Player.Movement
 {
-    public class PlayerMovement : MonoBehaviour
-    { 
-        [SerializeField, Range(1,10)] private float movementSpeed;
-        
-        private int _enumLength;
-        private int _gridIndex;
+    public class PlayerMovement : Framework.Movement
+    {
+        [SerializeField, Range(1,10)] private float movementSpeed; 
 
-        private void Awake()=> _enumLength = Enum.GetNames(typeof(GridPoints)).Length;
-        
-        private IEnumerator MoveTowardsGridPoint()
+        [SerializeField] private UnityEvent startedMoving = new();
+        private void Start()
```

Comment:
```suggestion
        [SerializeField] private UnityEvent onStartedMoving = new();
        [SerializeField] private UnityEvent onStopMoving = new();
                
        private void Start()
```
1. `startedMoving` => `onStartedMoving`
2. extra line tussen varible en functie
3. New onStopMoving event, dit kan goed gebruikt worden voor het starten van NPC ordering.